### PR TITLE
Test setup

### PR DESCRIPTION
**Pull request by Reza-rgb**
This will add all the support to create instrumented tests dealing with Hilt injection:
- `OverviewScreen` defines all the test tags we need to interact with Overview.
- `AssoDigestScreen` defines all the test tags we need to verify all the informations of a given association are correct.
- `OverviewTest` contains end-to-end tests for the Overview tab.
- `CustomTestRunner` is a necessary test runner for Hilt injection.

### CODE REVIEW

File: app/src/androidTest/java/com/swent/assos/OverviewTest.kt [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+      step("search a specific Association") {
+        assoListSearch {
+          performTextClearance()
+
+          performTextInput("180°C")
+        }
+      }
+
+      // Check if something is displayed when
+      // inputting "Challenge"
```
Comment (marinPh) [inline app/src/androidTest/java/com/swent/assos/OverviewTest.kt:65]: small typo

File: app/src/androidTest/java/com/swent/assos/screens/AssoDigestScreen.kt [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+class AssoDigestScreen(semanticsProvider: SemanticsNodeInteractionsProvider) :
+    ComposeScreen<AssoDigestScreen>(
+        semanticsProvider = semanticsProvider,
+        viewBuilderAction = { hasTestTag("AssoDigestScreen") }) {
+
+  val goBackButton: KNode = onNode { hasTestTag("GoBackButton") }
+
+  val assoTitle: KNode = onNode { hasTestTag("AssoTitle") }
+  // val assoDescription: KNode = onNode { hasTestTag("AssoDescription") }
```
Comment (marinPh) [inline app/src/androidTest/java/com/swent/assos/screens/AssoDigestScreen.kt:15]: delete commented code

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -1,3 +1,4 @@
 <resources>
     <string name="app_name">Assos</string>
```
Comment (marinPh) [inline app/src/main/res/values/strings.xml:2]: the project is called assos, i don't know if the capital A is an issue

**Review by marinPh — CHANGES_REQUESTED** [review]
Very good PR, very little changes needed.

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (Reza-rgb) [reply app/src/main/res/values/strings.xml:2]: The capital A is not an issue. You can see in a setting file that project name = "Assos"

**Review by marinPh — APPROVED** [review]
Good correction, did not need much anyways

**Pull request merged**